internal/pkg/instrumentation/bpf/google.golang.org/grpc/server/bpf/probe.bpf.c

```diff
@@ -167,3 +170,65 @@ int uprobe_http2Server_operateHeader(struct pt_regs *ctx)
 
     return 0;
 }
+
+// func (ht *http2Server) WriteStatus(s *Stream, st *status.Status)
+// https://github.com/grpc/grpc-go/blob/bcf9171a20e44ed81a6eb152e3ca9e35b2c02c5d/internal/transport/http2_server.go#L1049
+SEC("uprobe/http2Server_WriteStatus")
+int uprobe_http2Server_WriteStatus(struct pt_regs *ctx) {
+    struct go_iface go_context = {0};
+    get_Go_context(ctx, 2, stream_ctx_pos, true, &go_context);
+    void *key = get_consistent_key(ctx, go_context.data);
+
+    // Get parent context if exists
+    void *stream_ptr = get_argument(ctx, 2);
+    u32 stream_id = 0;
+    bpf_probe_read(&stream_id, sizeof(stream_id), (void *)(stream_ptr + stream_id_pos));
+    struct grpc_request_t *grpcReq = bpf_map_lookup_elem(&streamid_to_grpc_events, &stream_id);
+    if (grpcReq == NULL) {
+        // No parent span context, generate new span context
+        u32 zero = 0;
+        grpcReq = bpf_map_lookup_elem(&grpc_storage_map, &zero);
+        if (grpcReq == NULL) {
+            bpf_printk("failed to get grpcReq from storage map");
+            return -1;
+        }
+    }
+
+    void *grpcReq_event_ptr = bpf_map_lookup_elem(&grpc_events, &key);
+    // if grpcReq_event is null, then handleStream probe didn't run. Try starting a new span here
+    if (grpcReq_event_ptr == NULL)
+    {
+        grpcReq->start_time = bpf_ktime_get_ns();
+
+        start_span_params_t start_span_params = {
+            .ctx = ctx,
+            .sc = &grpcReq->sc,
+            .psc = &grpcReq->psc,
+            .go_context = &go_context,
+            // The parent span context is set by operateHeader probe
+            .get_parent_span_context_fn = dummy_extract_span_context_from_headers,
+            .get_parent_span_context_arg = NULL,
+        };
+        start_span(&start_span_params);
+
+        // Set attributes
+        if (!get_go_string_from_user_ptr((void *)(stream_ptr + stream_method_ptr_pos), grpcReq->method, sizeof(grpcReq->method)))
+        {
+            bpf_printk("Failed to read gRPC method from stream");
+            bpf_map_delete_elem(&streamid_to_grpc_events, &stream_id);
+            return 0;
+        }
+    }
+
+    void *status_ptr = get_argument(ctx, 3);
+    void *s_ptr = 0;
+    bpf_probe_read_user(&s_ptr, sizeof(s_ptr), (void *)(status_ptr + status_s_pos));
+    // Get status code from Status.s pointer
+    bpf_probe_read_user(&grpcReq->status_code, sizeof(grpcReq->status_code), (void *)(s_ptr + status_code_pos));
+
+    bpf_map_update_elem(&grpc_events, &key, grpcReq, 0);
+    bpf_map_delete_elem(&streamid_to_grpc_events, &stream_id);
+    return 0;
+}
+
+UPROBE_RETURN(http2Server_WriteStatus, struct grpc_request_t, grpc_events, events, 2, stream_ctx_pos, false)
```

Comment:
I am not sure about this return probe, since we already have `UPROBE_RETURN(server_handleStream...)`.
The `UPROBE_RETURN` macro sends the event to user-space so I don't think we need to have it twice.

Reply:
my thinking here was to use the return probe in the case where the initialization (handleStream) was missed. Since the handleStream entry probe deletes the event from the map on success, I think [this return probe will be a no-op](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/717ea4f1b7a8b60b0a4f0a99e1438adb9f1ad448/internal/include/uprobe.h#L33-L36) if we're not making a span for WriteStatus